src/main/scala/smile/infrastructure/DrawingSurface.scala

```diff
@@ -0,0 +1,326 @@
+package smile.infrastructure
+
+import smile.colors.{Color, PresetColor}
+import smile.infrastructure.Constants.MaximumOpacity
+import smile.pictures.{FillStyle, StrokeStyle}
+
+import java.awt.*
+import java.awt.geom.{AffineTransform, Arc2D, Ellipse2D, Path2D}
+import java.awt.image.BufferedImage
+
+/** Extension method for `Double` to truncate its value towards zero.
+  */
+extension (value: Double)
+  /** Truncates the decimal part of a double, moving towards zero.
+    *
+    * @return
+    *   The truncated `Double` value.
+    */
+  def truncate: Double =
+    if value > 0 then value.floor else value.ceil
+
+/** Represents a drawing surface for rendering shapes, images, and text. It encapsulates a
+  * `BufferAdapter` to provide a high-level drawing API.
+  *
+  * @param owner
+  *   The `BufferAdapter` instance that this `DrawingSurface` operates on.
+  */
+class DrawingSurface(val owner: BufferAdapter):
+
```

Comment:
I'm not entirely sure if I understand the difference in responsibilities between `DrawingSurface` and `BufferAdapter`. One is for drawing and other for seeing what has been drawn? Does that make sense?